client/src/components/cityview/realm/villagers/panels/discussion/DiscussionPanel.tsx

```diff
@@ -0,0 +1,173 @@
+  }, [realmEntityId]);
+
+  const { loreMachineJsonRpcCall, setIsDiscussionLoading } = useNpcStore();
+
+  const setSelectedDiscussionFromDirection = (direction: number) => {
+    const newTimeStamp = getNewDiscussionTimeStampFromDirection(selectedDiscussion, direction, LOCAL_STORAGE_ID);
+
+    if (newTimeStamp == -1) {
+      return;
+    }
+
+    setLastMessageDisplayedIndex(0);
+    setSelectedDiscussion(newTimeStamp);
+  };
+
+  const treatGenerateDiscussionResponse = (response: DiscussionRpcResponse) => {
+    setLastMessageDisplayedIndex(0);
+    const discussionTimeStamp = addDiscussionToStorage(response, LOCAL_STORAGE_ID);
+    setSelectedDiscussion(discussionTimeStamp);
+    setIsDiscussionLoading(false);
+  };
+
+  const gatherVillagers = async () => {
+    setIsDiscussionLoading(true);
+    try {
+      let response = await loreMachineJsonRpcCall("generateDiscussion", {
+        realm_id: Number(realmId!),
+        realm_entity_id: Number(realmEntityId!),
+        order_id: realm!.order,
+        user_input: DiscussionInput,
+      });
+      response = keysSnakeToCamel(response);
+      treatGenerateDiscussionResponse(response.discussion as DiscussionRpcResponse);
+    } catch (e) {
+      console.log(e);
+    }
+    setIsDiscussionLoading(false);
+  };
+
+  useEffect(() => {
+    const mostRecentTs = getMostRecentStorageDiscussionKey(LOCAL_STORAGE_ID);
+    if (mostRecentTs == -1) {
+      setSelectedDiscussion(null);
+      return;
+    }
+    setSelectedDiscussion(mostRecentTs);
+  }, [realmId]);
+
+  const handleUserMessageChange = (inputValue: string) => {
+    if (inputValue.length <= MAX_DISCUSSION_INPUT_LENGTH) {
+      setDiscussionInput(inputValue);
+    }
+  };
+
+  const setDate = (): string => {
+    const date = new Date(0);
+    date.setUTCSeconds(selectedDiscussion!);
+    return `${date.toLocaleDateString()} - ${date.toLocaleTimeString()}`;
+  };
+
+  return (
+    <div className="flex flex-col h-[250px] relative">
+      <div className="flex flex-row w-[100%] items-center justify-between" style={{ position: "relative", top: "2%" }}>
+        {selectedDiscussion! && (
+          <div className="flex relative">
+            <Button onClick={() => setSelectedDiscussionFromDirection(-1)}>
+              <ArrowPrev />
+            </Button>
+            <div className="text-white">{setDate()}</div>
+            <Button onClick={() => setSelectedDiscussionFromDirection(+1)} className="mr-2">
+              <ArrowNext />
+            </Button>
+          </div>
+        )}
+      </div>
+      <NpcChat />
+
+      <div className="flex my-1 items-center">
+        <TextInput
+          className="mx-2 border border-gold !text-white/70"
+          placeholder="Write something..."
+          value={DiscussionInput}
+          onChange={handleUserMessageChange}
+        />
+        <Button
+          className="mr-2"
+          onClick={gatherVillagers}
+          // variant={isDiscussionLoading ? "default" : "primary"}
```

Comment:
Remove this if not used anymore. Also I guess the `isDiscussionLoading` variable can be removed from this files as well then?

Reply:
Comment removed. The variable has already been removed. We do need the setIsDiscussionLoading however.